create filter

Lyzhenkoff opened the pull request:
добавлен фильтр Gain (Усилитель)

Lyzhenkoff committed:
create filter

add Gain (Усилитель)

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,7 @@
+const gainNode = context.createGain();
```
давай добавим класс-обертку

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,7 @@
+const gainNode = context.createGain();
+gainNode.gain.value = 0.4;
```
значение надо расхардкодить и передавать как параметр в конструктор класса или метод

Lyzhenkoff committed:
add class gain

Lyzhenkoff committed:
update filter.js

add a new filter(delay)

Lyzhenkoff committed:
change comments

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,56 @@
+/** Class create a new filter(gain). */
+class Gains {
+  createNode() {
```
методы не задокументированы, смотри usejsdoc.org

neSpecc replied:
вернее доки не там находятся

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,56 @@
+/** Class create a new filter(gain). */
```
надо написать, что делает этот фильтр

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,56 @@
+  }
+}
+
+/** Class create a new filter(delay). */
```
надо написать, что делает этот фильтр

Lyzhenkoff committed:
update JSdoc and create class filters

Lyzhenkoff committed:
create a new class ChanmelMerger

Lyzhenkoff committed:
update channelMerger and create splitter

Lyzhenkoff committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
divided into two classes

Lyzhenkoff committed:
update index.js

create only one class with all fiters

neSpecc reviewed (changes requested):
Давай сделаем абстрактный класс Filter и отнаследуем от него все фильтры.

filter.ts
```
class Filter {}
```

filter-gain.ts
```
class GainFilter extends Filter {}
```

filter-noise.ts
```
class NoiseFilter extends Filter {}
```

весь общий код будет содержаться в абстрактном классе Filter, а весь специфичный в потомках.

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,109 @@
+/**
+ * @ allFiltrers
```
в описании класса надо написать, зачем он нужен

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,109 @@
+class Filters {
+  createGain() {
+    /**
+     * Create a  gain filter.
```
описание метода должно быть над объявлением метода, читай usejsdoc

neSpecc commented on filters.js:
```diff
@@ -0,0 +1,109 @@
+     * Class create a new filter(delay).This module allows you to delay the sound for a certain time.
+     */
+    const delayNode = context.createDelay();
+    delayNode.delayTime.value = 2;
```
что такое 2? погугли magic number

neSpecc replied:
там и дальше по коду такие магические числа

Lyzhenkoff committed:
create filter.ts

create a new filters:delay,gain

Lyzhenkoff committed:
create new fiter

create high pass filter

nikmel2803 reviewed (commented):
- папку `.idea` коммитить вообще не надо (притяни мастера, там уже прописано нужное правило для игнора этой папки)
- `package-lock.json` не нужен, потому что мы используем `yarn` вместо `npm`
А почему тут вообще лежат js файлы? Проект же на TS

Lyzhenkoff committed:
delete js files and create filter Gain in filter.ts

delet filter.js,filters.js
add filter.ts
add gain filter

Lyzhenkoff committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
delete package-lock.json

delete file.json

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,19 @@
+class Filter {
```
надо разделить эти классы на два файла. В первом (filter.ts) будет абстрактный класс Filter, а во втором (filter-gain.ts) его потомок GainFilter

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,19 @@
+  }
+}
+
+const gainFilter: Gain = new Gain();
```
это для чего? лучше просто экспортировать класс. Конструировать его будет кто-то на уровне выше

neSpecc commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+
```
эти файлы надо убрать из реквеста

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,19 @@
+   * Create a  gain filter.
+   * Class create a new filter(gain).The module allows you to change the sound signal level.
+   */
+  constructor(gainNode: any, source: any, destination: any) {
```
параметры не задокументированы

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,19 @@
+   * Class create a new filter(gain).The module allows you to change the sound signal level.
+   */
+  constructor(gainNode: any, source: any, destination: any) {
+    this.gainNode.gain.value = 0.4;
```
свойства класса надо явно объявить и задокументировать

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,19 @@
+    this.source.connect(gainNode);
+    gainNode.connect(destination);
+    source.start(0);
+    super();
```
родительский класс лучше вызвать в самом начале конструктора

Lyzhenkoff committed:
create a new filter-gain.ts and change filter.ts

Lyzhenkoff committed:
delete .gitignore

Lyzhenkoff committed:
Delete .DS_Store file

Lyzhenkoff committed:
update filter-gain.ts

documented parameters
create comments

neSpecc reviewed (changes requested):
- код не встроен в структуру проекта
- параметры классов не объявлены
- абстрактный класс не задокументирован

Lyzhenkoff committed:
import filter to filter-gain.ts and comment

import filter to filter-gain.ts
export filter from filter.ts
comment class Filter

Lyzhenkoff committed:
update jsdoc comment

update  jsdoc comment in filter-gain.ts and filter.ts

PolinaShneider commented on filter-gain.ts:
```diff
@@ -0,0 +1,19 @@
+  constructor(gainNode: any, source: any, destination: any) {
+    /**
+     * @param gainNode - new gain object
+     * @param source - source of melody
```
Лучше задокументируй эти параметры перед строчкой `constructor(gainNode: any, source: any, destination: any) {`

WebStorm тебе сам заготовку для JSDoc сгенерировать должен. Параметр `destination` не описан. И все ли они типа `any`, или это пока промежуточная версия?

Lyzhenkoff committed:
build webpack and comment constructor

neSpecc commented on filter.ts:
```diff
@@ -0,0 +1,5 @@
+/**
+ * Create class Filter.
```
не понятное описание

neSpecc commented on filter-gain.ts:
```diff
@@ -0,0 +1,20 @@
+   */
+  constructor(gainNode: any, source: any, destination: any) {
+    super();
+    this.gainNode.gain.value = 0.4; // value 0..1 (can be changed dynamically)
```
свойства не объявлены

neSpecc commented on filter-gain.ts:
```diff
@@ -0,0 +1,20 @@
+ * Create a  gain filter.
+ * Class create a new filter(gain).The module allows you to change the sound signal level.
+ */
+class Gain extends Filter {
```
```suggestion
class GainFilter extends Filter {
```

neSpecc commented on filter-gain.ts:
```diff
@@ -0,0 +1,20 @@
+   * @param source - source of melody
+   * @param destination - place of destination
+   */
+  constructor(gainNode: any, source: any, destination: any) {
```
почему везде `any`?

Lyzhenkoff committed:
update JSDoc

Lyzhenkoff committed:
.DS_Store banished!

Lyzhenkoff committed:
add filters to folder Filter

Lyzhenkoff committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
update JSDoc

class properties declared

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,30 @@
+ */
+class GainFilter extends Filter {
+  /**
+   * new gain object
```
```suggestion
   * New gain object
```

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,30 @@
+  public gainNode: any;
+
+  /**
+   * source of melody
```
не понятное описание, которое дублирует название. + надо писать с большой буквы

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,30 @@
+   */
+  public source: any;
+  
+  constructor(gainNode: any, source: any, destination: any) {
```
параметры не задокументированы

Lyzhenkoff committed:
update JSDoc

Lyzhenkoff committed:
change tab

Lyzhenkoff committed:
update JSDoc

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,29 @@
+  /**
+   * New gain object
+   */
+  public gainNode: any;
```
уверен, что эти свойства должны быть видны снаружи?

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,29 @@
+   * Sound knot
+   */
+  public source: any;
+  /**
```
```suggestion

  /**
```

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,29 @@
+   */
+  constructor(gainNode: any, source: any, destination: any) {
+    super();
+    this.gainNode.gain.value = 0.4; // value 0..1 (can be changed dynamically)
```
value логично было бы вынести в свойство. Тогда у фильтра были бы методы по изменению этого значения

Lyzhenkoff committed:
update JSDoc and now volume - property class

Lyzhenkoff committed:
update filter

now gainNode is private properties

Lyzhenkoff committed:
value hardcoded

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,35 @@
+  /**
+   *  Loudness sound
+   */
+  public valueMelody: number;
```
почему melody?

Lyzhenkoff committed:
chage valueMelody to value

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,35 @@
+  /**
+   * Sound knot
+   */
+  public source: any;
```
не хватает обоснования для выбора модификаторов доступа

Lyzhenkoff committed:
update filter-gain.ts

all class properties are rewritten to private

Lyzhenkoff committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
export filter-gain to track

Lyzhenkoff committed:
update filter.ts

Lyzhenkoff committed:
update Track.ts

edit value of track

PolinaShneider commented on src/modules/Track.ts:
```diff
@@ -25,10 +26,12 @@ export default class Track {
+  public constructor(instrument: Instrument, melody: Melody, gainFilter: GainFilter) {
     this.instrument = instrument;
     this.melody = melody;
+    this.gainFilter = GainFilter;
```
У тебя есть свойство класса `gainFilter`?

PolinaShneider commented on src/modules/Track.ts:
```diff
@@ -47,13 +50,15 @@ export default class Track {
       }
     );
     this.instrument.stop(timeOffset);
+    this.gainNode.value = 1;
```
То же самое. Не вижу свойства класса `gainNode`. `Track` вроде ни от чего не отнаследован

Lyzhenkoff committed:
add gainNode filter

Lyzhenkoff committed:
Merge branch 'filters' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
add JSDoc

neSpecc commented on src/modules/Track.ts:
```diff
@@ -20,15 +21,20 @@ export default class Track {
+  /**
+   * Class create a new filter(gain).The module allows you to change the sound signal level.
+   */
+  private gainFilter: GainFilter;
```
это только один из доступных фильтров. Почему он является приватным свойством класса трэка? Фильтров может быть 100 разных штук

neSpecc commented on src/modules/filters/filter.ts:
```diff
@@ -0,0 +1,6 @@
+ * Create class Filter.(Using filters, we create high-level, configurable, and ready-to-use modules.
+ * These are amplifiers, delay lines, filters, convolution modules, splitters and mergers)
+ */
+export default abstract class Filter {
```
в абстрактый класс надо перенести общие свойства и методы, которые будут у всех фильтров

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+  public value: number;
+
+  /**
+   * @param gainNode - new gain object
```
не понятное описание

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+
+  /**
+   * @param gainNode - new gain object
+   * @param source - melody knot
```
не понятное описание

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+  /**
+   * @param gainNode - new gain object
+   * @param source - melody knot
+   * @param destination - place of purpose
```
не понятное описание

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+   * @param source - melody knot
+   * @param destination - place of purpose
+   */
+  constructor(gainNode: any, source: any, destination: any) {
```
типы не заданы

neSpecc replied:
я про any

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+   */
+  constructor(gainNode: any, source: any, destination: any) {
+    super();
+    this.gainNode.gain.value = this.value; // value 0..1 (can be changed dynamically)
```
как изменить значение? должен быть какой-то публичный метод?

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,37 @@
+    this.gainNode.gain.value = this.value; // value 0..1 (can be changed dynamically)
+    this.source.connect(gainNode); // connect source of melody to object(gainNode)
+    gainNode.connect(destination); // connect with place of destination
+    source.start(0);
```
уверен, что старт надо делать в конструкторе класса, а не в отдельном методе?

Lyzhenkoff committed:
update filters

add filter.ts

Lyzhenkoff committed:
update filters

add jsdoc

Lyzhenkoff committed:
delete filter 2

Lyzhenkoff committed:
delete filter-gain 2

neSpecc commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+   * @param source - interfaces allow you to add audio spatialization panning effects to your audio
+   * @param destination - once you are done processing your audio, these interfaces define where to output it.
+   */
+  constructor(gainNode: any, source: any, destination: any) {
```
эти узлы точно надо создавать снаружи и передавать в конструктор? это не относится к бизнес-логике самого фильтра?

Lyzhenkoff committed:
add type of class

Lyzhenkoff committed:
Merge branch 'filters' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
update filter.ts

added space

Lyzhenkoff committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

Lyzhenkoff committed:
Create bundle.min.js

Lyzhenkoff committed:
delete dist

Lyzhenkoff committed:
add index.js

ilyamore88 commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+   * @param destination - once you are done processing your audio, these interfaces define where to output it.
+   */
+  constructor(gainNode: GainNode, source: AudioBufferSourceNode, destination: AudioNode) {
+    super();
```
Конструктор родителя принимает на вход параметр, тут же ты его не передаёшь

ilyamore88 commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+  /**
+   * Interface represents a change in volume
+   */
+  private gainNode: any;
```
```suggestion
  private gainNode: GainNode;
```

ilyamore88 commented on src/modules/Track.ts:
```diff
@@ -20,15 +21,20 @@ export default class Track {
+  public constructor(instrument: Instrument, melody: Melody, gainFilter: GainFilter) {
     this.instrument = instrument;
     this.melody = melody;
+    this.gainFilter = GainFilter;
```
```suggestion
    this.gainFilter = gainFilter;
```

ilyamore88 commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+    super();
+    this.gainNode.gain.value = this.value; // value 0..1 (can be changed dynamically)
+    this.source.connect(gainNode); // connect source of melody to object(gainNode)
+    gainNode.connect(destination); // connect with place of destination
```
Давай заменим здесь однострочные комментарии на комментарии вида 
```
/**
 *
 */
```

ilyamore88 commented on src/modules/Track.ts:
```diff
@@ -55,6 +61,7 @@ export default class Track {
       }
     );
     this.instrument.stop(timeOffset);
+    this.gainFilter.value = 1;
```
Нелогично выставлять значение у `GainFilter` после выключения инструмента. Я бы добавил для этого публичный set у трека

ilyamore88 replied:
И присваивание какого-то значения громкости должно происходить либо в методе `configure` с попощью вызова соответствующего сеттера у `GainFilter`, либо передаваться в конструктор при создании `GainFilter`

ilyamore88 commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+   */
+  constructor(gainNode: GainNode, source: AudioBufferSourceNode, destination: AudioNode) {
+    super();
+    this.gainNode.gain.value = this.value; // value 0..1 (can be changed dynamically)
```
Нет смысла от такого присваивания в конструкторе, лучше добавь для фильтра сеттер, а в конструктор передавай какое-то дефолтное значение громкости

ilyamore88 commented on src/modules/filters/filter-gain.ts:
```diff
@@ -0,0 +1,36 @@
+  /**
+   * Interfaces allow you to add audio spatialization panning effects to your audio
+   */
+  private source: any;
```
```suggestion
  private source: OscillatorNode;
```

ilyamore88 reviewed (changes requested):
Добавь в метод `configure` класса `Track` подключение `gainFilter`, чтобы его можно было потестить.
Переименуй свои файлы в соответствии с остальными: `filter.ts` на `Filter.ts`, `filter-gain.ts` на `GainFilter.ts`.

Lyzhenkoff committed:
create a new filter

create lowpass filter

Lyzhenkoff committed:
create lowpassfilter

create lowpassfilter
add in instrument

Lyzhenkoff committed:
update instruments

Lyzhenkoff committed:
update track

Lyzhenkoff committed:
delete gain-filter

Lyzhenkoff committed:
delete gain-filter

Lyzhenkoff committed:
update low-pass filter

Lyzhenkoff committed:
add filter-lowpass

Lyzhenkoff committed:
add JSDoc

Lyzhenkoff committed:
create low pass filter

Lyzhenkoff committed:
connect filter with demo

Lyzhenkoff committed:
update JSDoc

ilyamore88 committed:
rename file to PascalCase, add typing and fix docs

ilyamore88 committed:
Merge branch 'master' of https://github.com/codex-team/codex.music into filters

 Conflicts:
	dist/index.js
	src/modules/instruments/Instrument.ts
	src/modules/instruments/SineWaveInstrument.ts

ilyamore88 committed:
rename file to PascalCase, fix type and docs

ilyamore88 committed:
connect filter to destination

ilyamore88 committed:
connect filter node

ilyamore88 committed:
delete timeout from track

neSpecc commented on src/modules/instruments/SineWaveInstrument.ts:
```diff
@@ -16,7 +18,15 @@ class SineWaveInstrument extends Instrument {
+  setFilterLowPass() {
+    this.filter = new LowPassFilter();
+    this.volumeNode.connect(this.filter.filterNode);
+    this.filter.filterNode.connect(audioContextManager.getAudioContext().destination);
```
для этого в фильтрах есть метод connect

neSpecc commented on src/modules/Track.ts:
```diff
@@ -35,7 +35,7 @@ export default class Track {
    */
   private configure(): void {
-    this.instrument.node.connect(audioContextManager.getAudioContext().destination);
+    this.instrument.lastNode.connect(audioContextManager.getAudioContext().destination);
```
почему не instrumentSourceNode ?

ilyamore88 committed:
delete volumeNode from Instrument.ts

ilyamore88 committed:
delete setting frequency in Instrument.ts

ilyamore88 commented:
Closes #5
